scripts/bo_benchmarks/demonstrations/branin_campaign_demonstration.py

```diff
@@ -0,0 +1,358 @@
+#!/usr/bin/env python3
+"""
+Branin optimization campaign with 10 trials and iteration-by-iteration evaluation metrics calculation.
+This script demonstrates proper use of ax-platform with GP evaluation metrics calculated after each trial.
+"""
+
+import numpy as np
+import pandas as pd
+from ax.service.ax_client import AxClient, ObjectiveProperties
+from ax.modelbridge.cross_validation import cross_validate
+from ax.plot.diagnostic import interact_cross_validation_plotly
+from ax.modelbridge.generation_strategy import GenerationStrategy, GenerationStep
+from ax.modelbridge.registry import Models
+import matplotlib.pyplot as plt
+import torch
+from gpcheck.models import GPModel, GPConfig
+from gpcheck.metrics import loo_pseudo_likelihood, importance_concentration
+from sklearn.metrics import r2_score
+from scipy.stats import kendalltau
+
+obj1_name = "branin"
+
+
+def branin(x1, x2):
+    y = float(
+        (x2 - 5.1 / (4 * np.pi**2) * x1**2 + 5.0 / np.pi * x1 - 6.0) ** 2
+        + 10 * (1 - 1.0 / (8 * np.pi)) * np.cos(x1)
+        + 10
+    )
+    return y
+
+
+def interval_score(y_true, lower, upper, alpha=0.05):
+    """
+    Calculate interval score for uncertainty quantification evaluation.
+    
+    For 95% confidence intervals (alpha=0.05), the formula is:
+    f_interval = (upper - lower) + (2/alpha) * max(0, lower - y) + (2/alpha) * max(0, y - upper)
+    
+    Args:
+        y_true: True values
+        lower: Lower confidence bounds
+        upper: Upper confidence bounds 
+        alpha: Confidence level (0.05 for 95% CI)
+    
+    Returns:
+        Array of interval scores (lower is better)
+    """
+    width = upper - lower
+    below_penalty = np.maximum(0, lower - y_true)
+    above_penalty = np.maximum(0, y_true - upper)
+    return width + (2 / alpha) * below_penalty + (2 / alpha) * above_penalty
+
+
+def normal_95ci(mean, std):
+    """Convert mean and standard deviation to 95% confidence intervals."""
+    lower = mean - 1.96 * std
+    upper = mean + 1.96 * std
+    return lower, upper
+
+
+def calculate_ax_cross_validation_metrics(ax_client, trial_index):
+    """Calculate metrics using Ax's built-in cross validation with the main client."""
+    try:
+        # Skip if not enough data for cross-validation
+        if trial_index < 3:
+            return None, None
+        
+        # Check if client has a Bayesian model (not just Sobol)
+        if ax_client.generation_strategy.current_step_index == 0:
+            print("  Main client still in Sobol phase, no CV available yet")
+            return None, None
+            
+        # Get the model bridge from the generation strategy 
+        model_bridge = ax_client.generation_strategy.model
+        if model_bridge is None:
```

Comment:
The point of having two AxClient-s where one AxClient exists solely to get the CV R^2 and CV interval score and contains the exact same data as the "real" AxClient, but never gets used for generating next trials, was so that we could get a notion of "what if we *were* using a model" even if he "real" AxClient is still in the initialization phase. Probably the way to address this is to skip Sobol entirely within the second AxClient's generation strategy, and in the calculation of the metrics, manually skip until we get to three iterations. That way, we'll ensure that we go directly to model-based, and it will be OK that we're attaching and completing the trials manually.